tuc: add page (#15033)
tldr-pages/tldr: pages/common/tuc.md

# tuc

> Cut text (or bytes) where a delimiter matches, then keep the desired parts.

> > A more user-friendly and powerful version of cut with sensible defaults.

Reviewer: ```suggestion
> A more user-friendly and powerful version of `cut` with sensible defaults.
```

> More information: <https://github.com/riquito/tuc>.

- Cut and rearrange fields:

> `\n echo "foo bar baz" | tuc -d ' ' -f 3,2,1    # => bazbarfoo`

Reviewer 2: ```suggestion
`echo "foo bar baz" | tuc -d '{{ }}' -f {{3,2,1}}`
```

Why the `\n`?

> - Replace the delimiter with something else:
> 
> `echo "foo bar baz" | tuc -d ' ' -r ' ➡ '  # => foo ➡ bar ➡ baz`

Reviewer 2: ```suggestion
- Replace the delimiter `space` with an arrow:

`echo "foo bar baz" | tuc -d ' ' -r ' ➡ '`
```

> - Keep a range of fields:
> 
> `echo "foo bar    baz" | tuc -d ' ' -f 2:   # => bar    baz`

Reviewer 2: ```suggestion
- Keep a range of fields:

`echo "foo bar    baz" | tuc -d ' ' -f {{2:}}`
```

> - Cut using regular expressions:
> 
> `echo "a,b, c" | tuc -e '[, ]+' -f 1,3  # => ac`

Reviewer 2: ```suggestion
- Cut using regular expressions:

`echo "a,b, c" | tuc -e '{{[, ]+}}' -f {{1,3}}`
```

> - Emit JSON output:
> 
> `echo "foo bar baz" | tuc -d ' ' --json     # => ["foo","bar","baz"]`

Reviewer 2: ```suggestion
- Emit JSON output:

`echo "foo bar baz" | tuc -d '{{ }}' --json`
```
